tensorflow/lite/micro/tools/make/targets/cortex_m_generic_makefile.inc

```diff
@@ -0,0 +1,156 @@
+    TARGET_SPECIFIC_FLAGS += -D__DSP_PRESENT=1 -D__FPU_PRESENT=1 -D__VTOR_PRESENT=1 -D__FPU_USED=1
+    FLOAT=hard
+    TARGET_ARCH := cortex-m33
+    GCC_TARGET_ARCH := $(TARGET_ARCH)
+  else ifeq ($(TARGET),$(filter $(TARGET),cortex-m0-generic))
+    CORE=M0
+    ARM_LDFLAGS := -Wl,--cpu=Cortex-M0
+    TARGET_SPECIFIC_FLAGS +=
+    TARGET_ARCH := cortex-m0
+    GCC_TARGET_ARCH := $(TARGET_ARCH)
+  else ifeq ($(TARGET),$(filter $(TARGET),cortex-m3-generic))
+    CORE=M3
+    ARM_LDFLAGS := -Wl,--cpu=Cortex-M3
+    TARGET_SPECIFIC_FLAGS +=
+    TARGET_ARCH := cortex-m3
+    GCC_TARGET_ARCH := $(TARGET_ARCH)
+  else ifeq ($(TARGET),$(filter $(TARGET),cortex-m4-generic))
+    CORE=M4
+    ARM_LDFLAGS := -Wl,--cpu=Cortex-M4.no_fp
+    TARGET_SPECIFIC_FLAGS+=
+    TARGET_ARCH := cortex-m4
+    GCC_TARGET_ARCH := $(TARGET_ARCH)
+  else ifeq ($(TARGET),$(filter $(TARGET),cortex-m4+fp-generic))
+    CORE=M4
+    ARM_LDFLAGS := -Wl,--cpu=Cortex-M4
+    TARGET_SPECIFIC_FLAGS+=
+    FLOAT=hard
+    TARGET_ARCH := cortex-m4+fp
+    GCC_TARGET_ARCH := cortex-m4
+  else ifeq ($(TARGET),$(filter $(TARGET),cortex-m7+fp-generic))
+    CORE=M7
+    ARM_LDFLAGS := -Wl,--cpu=Cortex-M7
+    TARGET_SPECIFIC_FLAGS +=
+    FLOAT=hard
+    TARGET_ARCH := cortex-m7+fp
+    GCC_TARGET_ARCH :=  cortex-m7
+  else ifeq ($(TARGET),$(filter $(TARGET),cortex-m7-generic))
+    CORE=M7
+    ARM_LDFLAGS := -Wl,--cpu=Cortex-M7.no_fp
+    TARGET_SPECIFIC_FLAGS +=
+    TARGET_ARCH := cortex-m7
+    GCC_TARGET_ARCH := $(TARGET_ARCH)
+  else
+    $(error "$(TARGET) not supported")
+  endif
+
+  # Toolchain specfic flags
+  ifeq ($(filter armclang,$(ALL_TAGS)),armclang)
+    CXX_TOOL  := armclang
+    CC_TOOL   := armclang
+    AR_TOOL   := armar
+    LD        := armlink
+    FLAGS_ARMC = \
+      --target=arm-arm-none-eabi \
+      -mcpu=$(TARGET_ARCH)
+    ifeq ($(BUILD_TYPE), release)
+       FLAGS_ARMC += -Wno-unused-private-field
+    endif
+
+    CXXFLAGS += $(FLAGS_ARMC)
+    CCFLAGS += $(FLAGS_ARMC)
+    LDFLAGS += $(ARM_LDFLAGS)
+
+    # Arm Compiler will not link the Math library (see below), therefore we're filtering it out.
+    # See Fatal error: L6450U: Cannot find library m:
+    # "Arm Compiler is designed to run in a bare metal environment,
+    # and automatically includes implementations of these functions,
+    # and so no such flag is necessary."
+    # https://developer.arm.com/documentation/100891/0611/troubleshooting/general-troubleshooting-advice
+    MICROLITE_LIBS := $(filter-out -lm,$(MICROLITE_LIBS))
+  else ifeq ($(filter armgcc,$(ALL_TAGS)),armgcc)
+    ifeq ($(shell which arm-none-eabi-gcc),)
+      export PATH := $(MAKEFILE_DIR)/downloads/gcc_embedded/bin/:$(PATH)
+      $(eval $(call add_third_party_download,$(GCC_EMBEDDED_URL),$(GCC_EMBEDDED_MD5),gcc_embedded,))
+    endif
+
+    CXX_TOOL  := arm-none-eabi-gcc
+    CC_TOOL   := arm-none-eabi-gcc
+    AR_TOOL   := arm-none-eabi-gcc-ar
+    LD        := arm-none-eabi-ld
```

Comment:
For consistency, let's use the `TARGET_TOOLCHAIN_PREFIX`, similar to https://github.com/tensorflow/tensorflow/blob/37da1f0ee19568ffddbb1b58d0f37bf8844427a6/tensorflow/lite/micro/tools/make/targets/bluepill_makefile.inc#L6

Reply:
Done